custom_reg_form/management/commands/email_opt_in_list_to_s3.py

```diff
@@ -0,0 +1,257 @@
+            type=str,
+            help="specify which setting to use as an aws secret key",
+        )
+        parser.add_argument(
+            "--aws-profile", type=str, help="specify which aws profile to use"
+        )
+        parser.add_argument(
+            "--fields",
+            type=str,
+            nargs="+",
+            help="fields to include in the output, see docstring header for how to specify fields",
+        )
+        parser.add_argument(
+            "--usage", action="store_true", help="print the module docstring header"
+        )
+        parser.add_argument(
+            "--s3-bucket-name",
+            type=str,
+            help="aws s3 bucket name where the csv file will be saved.",
+        )
+        parser.add_argument(
+            "--s3-object-name",
+            type=str,
+            help="the name under which the output will be saved.",
+        )
+
+    def handle(self, *args, **options):
+
+        if options["usage"]:
+            print(__doc__)
+            return
+
+        # The csv writer and boto3 client require different file modes
+        binary_fd = (
+            tempfile.TemporaryFile(mode="w+b")
+            if options["use_temp_file"]
+            else io.BytesIO()
+        )
+        text_fd = io.TextIOWrapper(binary_fd, encoding="utf-8", write_through=True)
+
+        writer = csv.writer(text_fd)
+
+        cursor = connections[read_replica_or_default()].cursor()
+
+        query = construct_query(options["fields"])
+
+        cursor.execute(query)
+
+        while 1:
+            array = (
+                # using the default arraysize as described in:
+                # https://www.python.org/dev/peps/pep-0249/#fetchmany
+                cursor.fetchmany()
+            )
+            if not array:
+                break
+            for row in array:
+                if options["skip_when_null"] and None in row:
+                    continue
+
+                writer.writerow(row)
+
+        cursor.close()
+
+        binary_fd.seek(0)
+
+        aws_access_key_id = None
+        if options["aws_access_key_setting"]:
+            aws_access_key_id = getattr(settings, options["aws_access_key_setting"])
+        aws_secret_access_key = None
+        if options["aws_secret_key_setting"]:
+            aws_secret_access_key = getattr(settings, options["aws_secret_key_setting"])
+
+        session = boto3.Session(
+            profile_name=options["aws_profile"],
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+
+        s3_client = session.resource("s3")
+
+        try:
+            s3_client.Bucket(  # pylint: disable=no-member
+                options["s3_bucket_name"]
+            ).put_object(Key=options["s3_object_name"], Body=binary_fd)
+        except NoCredentialsError as error:
+            LOGGER.exception(error)
+
+        text_fd.close()
+        binary_fd.close()
+
+        return
```

Comment:
I think this `return` statement could be skipped

Reply:
Done !